extmod/modussl_mbedtls.c

```diff
@@ -197,6 +205,14 @@ STATIC mp_uint_t socket_read(mp_obj_t o_in, void *buf, mp_uint_t size, int *errc
     mp_obj_ssl_socket_t *o = MP_OBJ_TO_PTR(o_in);
 
     int ret = mbedtls_ssl_read(&o->ssl, buf, size);
+    if (ret == MBEDTLS_ERR_SSL_PEER_CLOSE_NOTIFY) {
+        // end of stream
+        return 0;
+    }
+    if (ret == MBEDTLS_ERR_SSL_WANT_READ) {
+        *errcode = EWOULDBLOCK;
+        return 0;
+    }
```

Comment:
But can't you just move the if-block that you add in this patch, move it to after the existing "if (ret >= 0)..." ?  Because MBEDTLS_ERR_SSL_WANT_READ is negative, it can safely go after the "if (ret >= 0)", no?  Then it would mirror socket_write.